### PR TITLE
Update dependencies and make them more generic. Remove Cargo.lock

### PR DESCRIPTION
**Pull request by kellpossible**
My hope is that this pull request will make it easier for people to use this library in their own code going into the future. 

+ Update to the latest version of `Yew`.
+ Use less specific dependency version numbers, so that if the dependency gets updated in a patch release (e.g. `0.1.1` -> `0.1.2`) you don't need to update this library for people to be able to use the updated dependency (which should only contain non-breaking changes anyway).
+ Libraries typically shouldn't include a lock file.

I haven't tested this yet on the front-end but it still compiles fine.

Not sure whether this should contain a version bump to `0.1` to `0.2` considering the major changes to dependencies...?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,26 +3,26 @@ name = "yew_webview_bridge"
 description = "2-way communcation bridge between web-view and yew"
 keywords = ["yew", "web-view"]
 license = "MIT OR Apache-2.0"
-version = "0.1.0"
+version = "0.2.0"
```
Comment (hobofan) [inline Cargo.toml]: Could you please remove the version bump? I'll bump it when I release a new version.

File: Cargo.toml [hunk 2]
```diff
@@ -3,26 +3,26 @@ name = "yew_webview_bridge"
...
 edition = "2018"
 repository = "https://github.com/hobofan/yew_webview_bridge"
 readme = "README.md"
 
 [dependencies]
 serde = { version = "1", features = ["derive"] }
-serde_json = { version = "1.0.52" }
-rand = { version = "=0.6.5", features = ["wasm-bindgen"] }
+serde_json = { version = "1.0" }
+rand = { version = "0.7", features = ["wasm-bindgen"] }
```
Comment (hobofan) [inline Cargo.toml]: If there was an exact version specified, I'm sure there was a good reason for it.

IIRC, newer versions of rand caused an error (visible on the console) when used in the browser (at least in the environment provided by webview on OSX).

Would be good if you can reverse that change, and I'll add a tracking issue for it.

**Review by hobofan — CHANGES_REQUESTED** [review]
Looks good overall! See my comments in the review.

File: Cargo.toml (hunk 2 above)
Comment (kellpossible) [reply Cargo.toml]: Okay, no worries, I'll put back the exact version for `rand`.

**Comment by kellpossible** [comment]
I'll try to make sure I do a test on Linux and Windows with the front-end before this pull request is ready to be merged.
I guess it would be good to have an example in this repo to quickly test.

**Review by hobofan — APPROVED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -11,18 +11,18 @@ readme = "README.md"
...
 # frontend
-dashmap = { version = "3.11.1", optional = true }
+dashmap = { version = "3.11", optional = true }
 stdweb = { version = "0.4.20", optional = true }
-wasm-bindgen = { version = "0.2.62", optional = true }
-wasm-bindgen-futures = { version = "0.4.12", optional = true }
-yew = { version = "0.14", default-features = false, features = ["std_web"], optional = true }
+wasm-bindgen = { version = "0.2", optional = true }
+wasm-bindgen-futures = { version = "0.4", optional = true }
+yew = { version = "0.17", default-features = false, features = ["std_web"], optional = true }
```
Comment (hobofan) [inline Cargo.toml:22]:
```suggestion
yew = { version = "0.17", package = "yew-stdweb", default-features = false, features = ["std_web"], optional = true }
```

**Review by hobofan — CHANGES_REQUESTED** [review]

**Review by hobofan — APPROVED** [review]

**Comment by hobofan** [comment]
I just updated to this branch in one of my private projects, and ran some tests on OSX. LGTM 👍 

Just ping me once you ran your tests, and I'll merge! :)

**Comment by kellpossible** [comment]
@hobofan I realised that my project depends on wasm-sys/wasm-bindgen and I'm currently looking into a way to port this to use that system too, so got bogged down with that before I could test it. Feel free to merge if you think it will be fine.

File: Cargo.toml (hunk 2 above)
Comment (kellpossible) [reply Cargo.toml]: hey @hobofan I'm getting the following stacktrace while experimenting on using the new `rand`:

```
[Error] Unhandled Promise Rejection: Error: Out of bounds memory access (near '...e__h0d03c6547e5e0185(arg0, arg1, addHeap...')
	<?>.wasm-function[<T as core::convert::Into<U>>::into::h8e2dc1dc8cfddf13] (data:text/html,%3C…%3E:271:135)
	<?>.wasm-function[ppv_lite86::generic::dmap2::h9a1309b08a243710]
	<?>.wasm-function[rand_chacha::guts::refill_wide::ha379d3985d906d93]
	<?>.wasm-function[<rand_chacha::chacha::ChaCha20Core as rand_core::block::BlockRngCore>::generate::h228564273ae9a3c9]
	<?>.wasm-function[<rand::rngs::adapter::reseeding::ReseedingCore<R,Rsdr> as rand_core::block::BlockRngCore>::generate::h6885e5a5fa188921]
	<?>.wasm-function[rand_core::block::BlockRng<R>::generate_and_set::h6505ddccc6a5f801]
	<?>.wasm-function[<rand_core::block::BlockRng<R> as rand_core::RngCore>::next_u32::h8b977afb2b63748d]
	<?>.wasm-function[rand::distributions::integer::<impl rand::distributions::Distribution<u32> for rand::distributions::Standard>::sample::hc3ae3caece8847d4]
	<?>.wasm-function[rand::Rng::gen::hb1a7fdc6a5707799]
	<?>.wasm-function[yew_webview_bridge::Message<T>::generate_subscription_id::h956e2c2444c72c04]
```

Was this the same issue you were encountering on OSX? Was it reported?

Comment (hobofan) [reply Cargo.toml]: It's been some time, but I don't think it was that error, but a more proper one, like the one when you run into a `unimplemented!()`/`unreachable!()`.

**Pull request merged**